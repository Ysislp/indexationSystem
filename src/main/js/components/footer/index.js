import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getAppVersion } from '../../actions/appInfoActions';
import Page from './page';

class Footer extends Component {

  componentDidMount() {
    this.props.getAppVersion();
  }

  render () {

    const { version } = this.props.version;

    return (
      <Page 
        version={version}
      />
    );
  }
}

const mapStateToProps =  (state) => {
  return {
    version: state.appInfo.version
  }
}

export default connect(mapStateToProps, { getAppVersion })(Footer);
