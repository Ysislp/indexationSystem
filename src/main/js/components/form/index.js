import React, { Component } from 'react';
import axios from 'axios';
import Page from './page';
import { localhostUrl } from '../../config/paths';
import { regExpUrl } from '../../constants';

class Form extends Component {

  constructor(props) {
    super(props);

    this.state = {
      word: '',
      url: '',
      waiting: false,
      validUrl: true
    };

    this.onChangeText = this.onChangeText.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.validateUrl = this.validateUrl.bind(this);
  }

  onChangeText (e) {
    const { id, value } = e.target;
    this.setState({ [id]: value });
  }

  validateUrl () {
    const regexUrl = new RegExp(regExpUrl);

    this.setState({
      validUrl: regexUrl.test(this.state.url)
    });
  }

  handleSubmit (e) {
    e.preventDefault();

    this.setState({ waiting: true });

    const data = JSON.stringify({
      word: this.state.word, 
      url: this.state.url
    });

    check(data)
      .then(res => {
        alert('The URL has been '+res.toUpperCase());
        
        this.setState({ 
          word: '',
          url: '',
          waiting: false 
        });
      })
  }

  render () {
    const { 
      word, 
      url, 
      waiting, 
      validUrl 
    } = this.state;

    return (
      <Page 
        onChangeText={this.onChangeText}
        handleSubmit={this.handleSubmit}
        validateUrl={this.validateUrl}
        word={word}
        url={url}
        waiting={waiting}
        validUrl={validUrl}
      />
    );
  }
}

function check (data) {
  return axios.post(localhostUrl+'/check', data, {
    headers: {
      'content-Type': 'application/json'
    }})
      .then(response => response.data.state)
      .catch(e => console.log(e))
}

export default Form;