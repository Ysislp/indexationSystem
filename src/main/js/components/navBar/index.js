import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getInstanceId } from '../../actions/appInfoActions';
import Page from './page';

class NavBar extends Component {

  componentDidMount() {
    this.props.getInstanceId();
  }

  render () {

    const { instanceId } = this.props.instanceId;

    return (
      <Page 
        instanceId={instanceId}
      />
    );
  }
}

const mapStateToProps =  (state) => {
  return {
    instanceId: state.appInfo.instanceId
  }
}

export default connect(mapStateToProps, { getInstanceId })(NavBar);
