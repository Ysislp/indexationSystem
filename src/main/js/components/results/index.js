import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getResults, deleteUrl } from '../../actions/resultsActions';
import Page from './page';

export class Results extends Component {

  constructor(props) {
    super(props);

    this.handleDelete = this.handleDelete.bind(this);
  }

  componentDidMount() {
    this.props.getResults();
  }

  handleDelete (url) {
    const data = JSON.stringify({
      url: url
    });

    this.props.deleteUrl(data);
  }

  render () {

    const { results, server } = this.props;

    return (
        <Page 
          results={results}
          server={server}
          handleDelete={this.handleDelete}
        />
    );
  }
}

const mapStateToProps =  (state) => {
  return {
    results: state.results.urls,
    server: state.results.server
  }
}

export default connect(mapStateToProps, {getResults, deleteUrl})(Results);
