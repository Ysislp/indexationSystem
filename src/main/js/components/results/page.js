import React, { Fragment } from 'react';
import CssBaseLine from '@material-ui/core/CssBaseline'; // quita conflictos de estilo
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import DeleteIcon from '@material-ui/icons/Delete';
import NavBar from '../navBar';
import { 
  TableRow, 
  TableContainer, 
  Table, 
  TableHead, 
  TableBody, 
  TableCell, 
  Button 
} from '@material-ui/core';


function Page (props) {
  const classes = useStyles();

  const {
    results,
    server,
    handleDelete
  } = props;

  const records = results.length !== 0;

  return (
    <div>
      <CssBaseLine />
      <NavBar />
      <TableContainer className={classes.paper} component={Paper}>
        <Table className={classes.table} size="small" aria-label="a dense table">
          <TableHead>
            <TableRow>
              <TableCell align="center">ID</TableCell>
              <TableCell align="center">URL</TableCell>
              <TableCell align="center">Actions</TableCell>
            </TableRow>
          </TableHead>
          {records ?
            <TableBody className="results-table" data-test="tableResults">
              {results.map((item) => (
                <TableRow key={item.id}>
                  <TableCell align="center">{item.id}</TableCell>
                  <TableCell>{item.url}</TableCell>
                  <TableCell align="center">
                    <Button 
                      color="secondary"
                      className={classes.button}
                      onClick={() => handleDelete(item.url)}
                      startIcon={<DeleteIcon />}
                      data-test="buttonDelete"
                    > 
                      Delete
                    </Button>
                  </TableCell>
                </TableRow>
              ))}
            </TableBody>
          :
            <TableBody className="empty-message" data-test="tableEmpty">
              <TableRow>
                <TableCell align="center" colSpan={3} >
                  No records available
                </TableCell>
              </TableRow>
            </TableBody>
          }
        </Table>
      </TableContainer>
      {server ? 
        <footer className={classes.footer}>
          By {server}
        </footer>
      :
        <footer className={classes.footer}></footer>
      }
    </div>
  );
}

export default Page;

const useStyles = makeStyles({
  table: {
    minWidth: 650,
    width: 200,
    marginTop: 64
  },
  paper: {
    display: 'flex',
    justifyContent: 'center',
    height: '100vh'
  },
  button: {
    margin: 10,
  },
  footer: {
    background: '#ffffff',
    padding: 20,
    textAlign: 'left',
    color: '#828282'
  }
});

