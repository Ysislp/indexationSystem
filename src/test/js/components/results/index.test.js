import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom'
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import ResultsApp from '../../../../main/js/components/results';
import { Results } from '../../../../main/js/components/results';
import Page from '../../../../main/js/components/results/page';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('Results component Test', () => {

  let store;
  let getResults;
  let deleteUrl;
  let getInstanceId;
  
  beforeEach(() => {

    store = mockStore({
      results: { 
        urls: [
          {
            id: 1,
            url: "http://google.com"
          },
        ]
      },
      appInfo: {
        instanceId: 'test',
        version: 'test',
      }
    });

    getResults = jest.fn();
    deleteUrl = jest.fn();
    getInstanceId = jest.fn();
  })

  it('Snapshot Results', () => {
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <ResultsApp />
        </MemoryRouter>
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Should onClick handler', () => {
    const results = [{id: 1, url:"http://google.com"}];
    const handleDelete = jest.fn();
    const component = shallow(
      <Page 
        handleDelete={handleDelete}
        results={results}
      />
    );
    const wrapper = component.find(`[data-test='buttonDelete']`);

    wrapper.simulate('click');
    expect(handleDelete).toBeCalled();
  });

  it('Should display TableBody with results', () => {
    const results = [{id: 1, url:"http://google.com"}];
    const component = shallow(
      <Page
        results={results}
      />
    );
    const wrapper = component.find(`[data-test='tableResults']`);

    expect(wrapper).toBeTruthy();
  });

  it('Should display TableBody with results', () => {
    const component = shallow(
      <Page
        results={[]}
      />
    );
    const wrapper = component.find(`[data-test='tableResults']`);
    const wrapper2 = component.find(`[data-test='tableEmpty']`);

    expect(wrapper).toEqual({});
    expect(wrapper2).toBeTruthy();
  });

  it('Should handleDelete()', () => {
    const data = JSON.stringify({
      url: "http://google.com"
    });

    const component = shallow(
      <Results 
        getResults={getResults}
        deleteUrl={deleteUrl}
      />
    );
    const instance = component.instance();
    

    instance.handleDelete(data);
    expect(deleteUrl).toBeCalled();
  });
  
});