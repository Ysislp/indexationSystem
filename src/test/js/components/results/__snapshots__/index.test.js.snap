// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component Test Snapshot Results 1`] = `
Array [
  <div
    className="makeStyles-root-4"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root makeStyles-menuButton-5"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
          />
        </svg>
        <h6
          className="MuiTypography-root makeStyles-title-6 MuiTypography-h6"
        >
          Indexation System
        </h6>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <a
              className="makeStyles-links-7"
              href="/all"
              onClick={[Function]}
            >
              Ver Todos
            </a>
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <a
              className="makeStyles-links-7"
              href="/"
              onClick={[Function]}
            >
              Agregar
            </a>
          </span>
        </button>
      </div>
    </header>
  </div>,
  <div
    className="MuiPaper-root MuiTableContainer-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      aria-label="a dense table"
      className="MuiTable-root makeStyles-table-1"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeSmall"
            scope="col"
          >
            ID
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeSmall"
            scope="col"
          >
            URL
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeSmall"
            scope="col"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root results-table"
        role={null}
      >
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
          >
            1
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            http://google.com
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-3 MuiButton-textSecondary"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <span
                  className="MuiButton-startIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </span>
                Delete
              </span>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
]
`;
